# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 100 # Hz
    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    servo_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
joint_state_broadcaster:
  ros__parameters:
    use_local_topics: false

arm_controller:
  ros__parameters:
    joints:
      - shoulder_pan
      - shoulder_tilt
      - elbow
      - wrist_pitch
      - wrist_roll
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    allow_nonzero_velocity_at_trajectory_end: true

servo_controller:
  ros__parameters:
    joints:
      - shoulder_pan
      - shoulder_tilt
      - elbow
      - wrist_pitch
      - wrist_roll
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    allow_nonzero_velocity_at_trajectory_end: true
    allow_partial_joints_goal: true
    allow_integration_in_goal_trajectories: true
